qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,311 @@
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> AdaptVQEResult:
+        """Computes the minimum eigenvalue.
+
+        Args:
+            operator: Operator whose minimum eigenvalue we want to find.
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than :class:`~VQE` or an ansatz other than
+                :class:`~.EvolvedOperatorAnsatz` is provided or if the algorithm finishes due
+                to an unforeseen reason.
+            AlgorithmError: If `quantum_instance` is not provided.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An :class:`~.AdaptVQEResult` which is a :class:`~.VQEResult` but also but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            termination criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
+            raise QiskitError("The AdaptVQE ansatz must be of the EvolvedOperatorAnsatz type.")
+
+        # Overwrite the solver's ansatz with the initial state
+        solver_ansatz = copy.deepcopy(self.solver.ansatz)
+        self._excitation_pool = copy.deepcopy(solver_ansatz.operators)
```

Comment:
We need a deep copy of the operators? We are deepcopying the ansatz, which may have operators set and deep copying operators too. Can we not simply say that the solver ansatz will be manipulated by AdaptVQE - do we really need to deep copy it. I understand we need the list of operators but again deepcopy...

Reply:
The `solver_ansatz` is only used to ensure that the internal `VQE` remains unchanged from the perspective of the external scope. This was suggested by @Cryoris [here](https://github.com/Qiskit/qiskit-terra/pull/7930#discussion_r928924813).
I am fine either way. I did however remove the second `deepcopy` as it is indeed not necessary :+1: